sdk/storage/azure-storage-quickquery/src/main/java/com/azure/storage/quickquery/BlobQuickQueryAsyncClient.java

```diff
@@ -4,10 +4,202 @@
+import com.azure.storage.blob.models.BlobRequestConditions;
+import com.azure.storage.blob.models.CpkInfo;
+import com.azure.storage.common.ParallelTransferOptions;
+import com.azure.storage.common.ProgressReporter;
+import com.azure.storage.common.implementation.Constants;
+import com.azure.storage.common.implementation.UploadBufferPool;
+import com.azure.storage.common.implementation.UploadUtils;
+import com.azure.storage.quickquery.models.BlobQuickQueryAsyncResponse;
+import com.azure.storage.quickquery.models.BlobQuickQueryDelimitedSerialization;
+import com.azure.storage.quickquery.models.BlobQuickQueryJsonSerialization;
+import com.azure.storage.quickquery.models.BlobQuickQuerySerialization;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.util.function.Tuples;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+
+import static com.azure.core.util.FluxUtil.byteBufferToArray;
+import static com.azure.core.util.FluxUtil.monoError;
+import static com.azure.core.util.FluxUtil.withContext;
 
 /**
- * TEMP
+ * This class provides a client that contains all operations that apply to Azure Storage Blob quick query.
+ *
+ * <p>This client offers the ability to query contents of a structured blob. </p>
+ *
+ * @see BlobQuickQueryClientBuilder
  */
 @ServiceClient(builder = BlobQuickQueryClientBuilder.class, isAsync = true)
 public class BlobQuickQueryAsyncClient {
+
+    private final ClientLogger logger = new ClientLogger(BlobQuickQueryAsyncClient.class);
+
+    private final AzureBlobStorageImpl client;
+    private final CpkInfo customerProvidedKey;
+
+    /**
+     * Package-private constructor for use by {@link BlobQuickQueryClientBuilder}
+     *
+     * @param pipeline The pipeline used to send and receive service requests.
+     * @param url The endpoint where to send service requests.
+     * @param serviceVersion The version of the service to receive requests.
+     * @param customerProvidedKey Customer provided key used during encryption of the blob's data on the server, pass
+     * {@code null} to allow the service to use its own encryption.
+     */
+    BlobQuickQueryAsyncClient(HttpPipeline pipeline, String url, BlobServiceVersion serviceVersion,
+        CpkInfo customerProvidedKey) {
+        this.client = new AzureBlobStorageBuilder()
+            .pipeline(pipeline)
+            .url(url)
+            .version(serviceVersion.getVersion())
+            .build();
+        this.customerProvidedKey = customerProvidedKey;
+    }
+
+    /**
+     * Queries the entire blob. NOTE: Returns raw avro.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.quickquery.BlobQuickQueryAsyncClient.query#String}
+     *
+     * @param expression The query expression.
+     * @return A reactive response containing the queried data.
+     */
+    public Flux<ByteBuffer> query(String expression) {
```

Comment:
for preview I can make these package private - eventually we want these methods to work though